### PR TITLE
[FEAT] Database Setup - Models, and Seeding

### PR DESCRIPTION
**Pull request by Micah-Shallom**
@NwokoyeChigozie 

Setup Database initialization and seeding 

https://github.com/hngprojects/hng_boilerplate_golang_web/issues/42#issue-2420187585

### CODE REVIEW

File: internal/models/organisation.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+	// Add user to organisation
+
+	err := db.Model(user).Association("Organisations").Append(orgs...)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (NwokoyeChigozie) [inline internal/models/organisation.go:28]: This should have been done with a receiver
func (u *User)AddToOrganisation(db *gorm.DB, orgs []interface{})
is not a bad place to start. and this should be in the user.go

File: services/seed/seed.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	userEmail := ""
+
+	if userID == 1 {
+		userEmail = "john@example.com"
+	} else if userID == 2 {
+		userEmail = "jane@example.com"
+	} else {
+		return userResp, errors.New("user id does not exist")
```
Comment (NwokoyeChigozie) [inline services/seed/seed.go:33]: I don't know why you are doing this here, but it is very wrong,
if this user exists in the db, why not use the actual record id to find the record? why hardcode this here?

File: services/seed/seed.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+	"github.com/hngprojects/hng_boilerplate_golang_web/internal/models"
+)
+
+func GetUser(c *gin.Context, db *gorm.DB) (models.User, error) {
+
+	var userResp models.User
+
+	// Get the user_id from the URL
+	userIDStr := c.Param("user_id")
```
Comment (NwokoyeChigozie) [inline services/seed/seed.go:18]: get this from the controller, not the service

File: services/seed/seed.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+	if userID == 1 {
+		userEmail = "john@example.com"
+	} else if userID == 2 {
+		userEmail = "jane@example.com"
+	} else {
+		return userResp, errors.New("user id does not exist")
+	}
+
+	if err := db.Preload("Profile").Preload("Products").Preload("Organisations").Where("email = ?", userEmail).First(&userResp).Error; err != nil {
```
Comment (NwokoyeChigozie) [inline services/seed/seed.go:36]: this should be in the user model

**Review by NwokoyeChigozie — CHANGES_REQUESTED** [review]

**Comment by Micah-Shallom** [comment]
@NwokoyeChigozie ... Changes has been affected Sir...Kindly review modifications.

**Review by NwokoyeChigozie — APPROVED** [review]

**Comment by Micah-Shallom** [comment]
This PR was worked on by @Micah-Shallom slackID- stonepillar  and @Cyberguru1 slackID- cyberguru

**Pull request merged**